### ISSUE TITLE
Colors Hex - Incorrect

### ISSUE THREAD

**Issue by xiaosongshu** [mac-cain13/R.swift#301]
Would there be any reason my colors from my .clr file would appear in R.generated incorrectly? 

For example, in Sketch, I'm saving "#F2F6F7" in my .clr file and in R.generated, I have:

```
/// <span style='background-color: #EFF4F5; color: #100B0A; padding: 1px 3px;'>#EFF4F5</span> White smoke
      static let whiteSmoke = Rswift.ColorResource(name: "White smoke", red: 0.935933113098145, green: 0.955909192562103, blue: 0.960774302482605, alpha: 1.0)
``` 

`#EFF4F5` is VERY close to `#F2F6F7`, but not exact. This is happening with all of my colors. Am I doing something wrong?

**Comment by tomlokhorst** [comment]
I'm unable to reproduce this in the quick test I just did.
If I create a .clr file from the system color picker with a `#F2F6F7` color, R.swift will interpret it as such.

How are you creating a .clr file from Sketch? I looked around and couldn't find a way to do that.

I did notice that R.swift currently always assumes a RGB color space, that might be incorrect on macOS and iOS 10 that support more color spaces. So maybe that is part of what's happening here.

**Comment by renrawnalon** [comment]
It looks like what is happening is, when you are defining your colors, you are using the sRGB color space (which might be the default for Sketch, I'm not sure though), but R.swift expects colors to be defined in the Generic RGB color space. This is more of an issue with the format colors are saved in than an issue of R.swift generating the wrong colors.

I was able to reproduce the conversion from `#F2F6F7` to `#EFF4F5` in Xcode by doing the following:

1. Open the Color Picker in Xcode (through storyboard or anything else).
2. Select the second tab and choose RBG Slider.
3. Click the settings button and select `sRGB`.
4. Enter `#F2F6F7` into the Hex Color field and press Enter.
    <img width="512" alt="srgb" src="https://cloud.githubusercontent.com/assets/605237/23327858/4967dbb0-fb57-11e6-928c-b74d80c80cca.png">
5. Click the settings button again and this time, select `Generic RGB`.
    <img width="513" alt="after changing to generic rgb" src="https://cloud.githubusercontent.com/assets/605237/23327860/49b7d69c-fb57-11e6-945a-a52951a06e47.png">

You should see that the Hex value changes to `#EFF4F5`. This conversion happens implicitly when you save to your .clr file, so R.swift ends up generating colors for the converted values.

To fix this, with `Generic RGB` selected, reentering `#F2F6F7 ` into the Hex Color field and pressing enter should give the correct color in the color space that R.swift is expecting.

<img width="510" alt="proper color setting in generic rgb" src="https://cloud.githubusercontent.com/assets/605237/23327859/49b5ef80-fb57-11e6-9758-69d94b3053d2.png">

You can re-save that new value to "White smoke" in your .clr file, and R.swift should generate the proper hex value for you.

Make sure you don't forget to press enter each time you update the Hex Color field, or Xcode won't register the change when you go to add your colors to your .clr file. Also, you'll need to copy the .clr file into your project so that R.swift can find it, if that's how you have R.swift setup.

I don't know if you can do something similar to changing the color space in Sketch, but as a second best option, if you follow the steps above, you should because to generate the proper colors for R.swift through Xcode. If this solves your problem, let me know and I'll submit a PR to update the documentation making this clear. If you are able to solve your problem using Sketch, I think that would be useful information to add to the documentation as well.

I hope this helps!

**Comment by xiaosongshu** [comment]
Awesome, thanks for the help guys!

@renrawnalon, I found these two links after digging more into this:
https://www.sketchapp.com/learn/documentation/other/color-management/
https://sketchtalk.io/discussion/2247/sketch-doesnt-do-color-management-and-why-thats-a-bad-idea

The second link provided the solution to making sure Sketch synced up correctly with Xcode. In short, Sketch was saving the colors in reference to my monitor's color gamut. It was set on a color setting that was effectively sRGB. By changing my monitor's display profile to `Generic RGB Profile` and rebuilding each of my saved colors in Sketch that I used for my .clr file with the new hex values like you mentioned, everything translates correctly once added to Xcode. Of course, this solution might not work for those who calibrate their monitors, but changing the color profile made a negligible difference for me personally.

**Comment by renrawnalon** [comment]
@xiaosongshu I'm glad you were able to find a solution within Sketch. It would have been really inconvenient if you had to go through Xcode to generate proper rgb values for your .clr file.

@tomlokhorst @mac-cain13 
Two questions.

1. Do you think that the solution to this problem is worth mentioning somewhere in the documentation or the FAQ? The reason I knew where to look for the solution is because I came across this same problem recently when using custom .clr files that I created in Xcode, and didn't even realize it until the designer on my team pointed it out to me. Others might be experiencing this same problem without realizing it as well.
If you'd like, I'd be happy to submit a PR to update the documentation with the information in this issue.

2. Given this added complication in using .clr files (properly) in R.swift, might it be worth reconsidering support of .txt or .json file formats as an alternative to .clr files, as mentions [here](https://github.com/mac-cain13/R.swift/issues/204)? I know that the issue of colors is on the very edge of the scope of R.swift, and generating proper .clr files is completely outside of scope, but specifying rgba values in text format would be a surefire way of avoiding the problem of this issue. It might be best to continue this discussion on the [original issue](https://github.com/mac-cain13/R.swift/issues/204) if you think it merits reconsideration.

**Comment by mac-cain13** [comment]
@renrawnalon Feel free to create a PR that explains this in de docs! Seems like something that would be nice to document somewheren.

I still have mixed feelings about color json/txt files, but I get your point. I'm also thinking about more of a "style" file where you also can predefine font styles for example. To make collaboration with designers easier. Again that is not really the scope. Maybe we should remove colors completely from R.swift and create another tool to do such a thing.

Have to think a little more about it.

**Comment by tomlokhorst** [comment]
With regards to the current situation:
I think the fact that R.swift always interprets a .clr file as RGB, is simply a bug.

On iOS 10 it should be possible to use other color spaces, so R.swift should output the correct code for that.
If we don't want to add that feature yet (or for iOS versions below 10), R.swift should refuse to process the .clr file, if it isn't in RGB, and emit a warning.

**Comment by mac-cain13** [comment]
This is fixed in the next release by PR #312, for P3 color support is a seperate ticket: https://github.com/mac-cain13/R.swift/issues/317
